File: frontend/types/listing-rentals.ts
export interface ListingRentalRequest {
  id: string;
  listing_id: string;
  renter_id: string;
  start_date: string;
  end_date: string;
  total_days: number;
  total_price: number;
  status: string;
  created_at: string;
  conversation_id: string | null;
}

export interface ListingRentalRequestPayload {
  start_date: string;
  end_date: string;
  total_days: number;
  total_price: number;
}

export interface ListingRentalMessage {
  id: string;
  conversation_id: string;
  sender_id: string;
  sender_name: string | null;
  body: string;
  created_at: string;
}

export interface ListingRentalConversationSummary {
  id: string;
  listing_id: string;
  listing_title: string;
  counterparty_name: string | null;
  role: "owner" | "renter";
  status: string;
  total_price: number;
  total_days: number;
  start_date: string;
  end_date: string;
  last_message: string | null;
  last_message_at: string | null;
  created_at: string;
}

File: backend/schemas/listing_rentals.py
from datetime import date, datetime
from decimal import Decimal
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class ListingRentalRequestCreate(BaseModel):
    start_date: date
    end_date: date
    total_days: int = Field(ge=1)
    total_price: float = Field(ge=0)


class ListingRentalRequestRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    listing_id: UUID
    renter_id: UUID
    start_date: date
    end_date: date
    total_days: int
    total_price: Decimal
    status: str
    created_at: datetime
    conversation_id: UUID | None = None


class ListingMessageCreate(BaseModel):
    body: str = Field(min_length=1, max_length=2000)


class ListingMessageRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    conversation_id: UUID
    sender_id: UUID
    sender_name: str | None = None
    body: str
    created_at: datetime


class ListingRentalConversationSummary(BaseModel):
    id: UUID
    listing_id: UUID
    listing_title: str
    counterparty_name: str | None = None
    role: str
    status: str
    total_price: Decimal
    total_days: int
    start_date: date
    end_date: date
    last_message: str | None = None
    last_message_at: datetime | None = None
    created_at: datetime
